text/tls-and-alpn.rst

```diff
@@ -0,0 +1,380 @@
+        "database": "edgedb",
+        "tls_private_key_password": "...."
+    }
+
+Question: I don't feel comfortable storing the key passphrase here in clear
+text, especially if this is in production environment and we asked the user
+for passphrase but stored in this file. Perhaps we could use this JSON value
+only for the development process, and introduce a more secure (thus less
+convenient) way to obtain the private key password, like a customizable command
+as in PostgreSQL?
+
+Under the new folder ``certs/``, we'll place all certificates and keys used by
+the EdgeDB instances. The cert files will be named after the instance names.
+For development scenario we'll likely generate those certs for the user, while
+in production we'll "import" the certs specified by the user, by copying and
+renaming the cert and key files to the ``certs/`` folder.
+
+
+Development Scenario
+^^^^^^^^^^^^^^^^^^^^
+
+From the user's perspective, ``edgedb server init`` (without parameters we are
+introducing in this RFC later) will now automatically generate a self-signed
+certificate with a random key passphrase, and set the above config properly.
+This is considered to be the development scenario, so we try not to let the
+user do extra work here - the ``edgedb server init`` command shouldn't ask
+questions about generating the self-signed certificate. Instead, it should warn
+about the self-signed certificate, and provide suggestions for production.
+
+Question: because the certificate is self-signed, certificate verification on
+the client side would fail. Now that we know we are "in development mode"
+(using self-signed certificate), we could leverage the tooling to trust the
+certificate on the client side, by adding a boolean flag in
+``$WELLKNOWN/credentials/<instance>.json``. We could update our client bindings
+and the CLI to honor this flag, load and trust the certificate so that the user
+doesn't have to write extra code in development. Though, this works only for
+local clients - remote clients don't have access to the local file system so
+there has to be a different way for them to trust the dev server. If this
+option is good, I will expand ths paragraph with details. Alternatives:
+
+1. Automatically build a local CA, and issue certificates from that CA. The
+   clients and CLI will only need to trust the CA certificate. This doesn't
+   solve the remote client trouble though.
+2. Provide a client-side switch that the user could easily set, so that the
+   user could explicitly turn it on for development mode, and off for
+   production.
+
+
+Production Scenario
+^^^^^^^^^^^^^^^^^^^
+
+Question: do we want to manage the certificate for the user? That means to
+import the user certificate files and store locally, warn about expiry or even
+run certbot to renew some free certs (this might be too much). If yes, the
+following in this section is just one of the options.
+
+``edgedb server init`` will have additional parameters to import the TLS
```

Comment:
`server init` is not for production (at least yet). So I don't see why you would ever want to import certificates here. Just always use generated ones (until we have `server init --system` or something, if ever).